pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -846,6 +846,8 @@ func (pl *dynamicResources) PreScore(ctx context.Context, cs *framework.CycleSta
 		logger.V(5).Info("no pending claims", "pod", klog.KObj(pod))
 		return nil
 	}
+
+	nodes := framework.GetNodes(nodeInfos)
```

Comment:
This could be quite expensive in a big cluster. Instead, change `haveAllPotentialNodes` to undertand NodeInfos.